ml-agents/mlagents/trainers/rl_trainer.py

```diff
@@ -265,6 +269,14 @@ def add_rewards_outputs(
             "The add_rewards_outputs method was not implemented."
         )
 
+    def add_policy(self, brain_parameters: BrainParameters) -> None:
+        """
+        Adds policy to trainers list of policies
+        """
+        policy = self.create_policy(brain_parameters)
+        self.policy = policy
```

Comment:
This overwrites the old policy?

Reply:
In ppo/sac this should only be called once.  In the ghost trainer this will be overridden to add to self.policies.  We'll still need to set self.policy in the wrapped trainer though.

Reply:
If it's only meant to be called once, is that worth enforcing with an exception if self.policy is already set?